pkg/services/apiserver/aggregator/aggregator.go

```diff
@@ -74,13 +124,33 @@ func CreateAggregatorConfig(commandOptions *options.Options, sharedConfig generi
 	}
 
 	if err := commandOptions.AggregatorOptions.ApplyTo(aggregatorConfig, commandOptions.RecommendedOptions.Etcd, commandOptions.StorageOptions.DataPath); err != nil {
-		return nil, nil, err
+		return nil, nil, nil, err
+	}
+
+	// Exit early, if no remote services file is configured
+	if commandOptions.AggregatorOptions.RemoteServicesFile == "" {
+		return aggregatorConfig, sharedInformerFactory, nil, nil
 	}
 
-	return aggregatorConfig, sharedInformerFactory, nil
+	caBundlePEM, err := readCABundlePEM(commandOptions.AggregatorOptions.APIServiceCABundleFile, commandOptions.ExtraOptions.DevMode)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	remoteServices, err := readRemoteServices(commandOptions.AggregatorOptions.RemoteServicesFile)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	return aggregatorConfig, sharedInformerFactory, &RemoteServicesConfig{
```

Comment:
would it make sense to wrap the k8s aggregator config and add this as extra config?